library/socket/shared/pack_sockaddr.rb

```diff
@@ -17,6 +17,9 @@
 
     sockaddr_in = Socket.public_send(@method, nil, '127.0.0.1')
     Socket.unpack_sockaddr_in(sockaddr_in).should == [0, '127.0.0.1']
+
+    sockaddr_in = Socket.public_send(@method, 'http', '127.0.0.1')
+    Socket.unpack_sockaddr_in(sockaddr_in).should == [80, '127.0.0.1']
```

Comment:
Could you add it in a separate new `it`, like `it "resolves the service name to a port" do`?

In general one spec example should test one thing (the above example is not a great example).